src/main/java/cc/hyperium/gui/HyperiumTextField.java

```diff
@@ -0,0 +1,751 @@
+/*
+ *      Copyright (C) 2018  Hyperium <https://hyperium.cc/>
+ *
+ *      This program is free software: you can redistribute it and/or modify
+ *      it under the terms of the GNU Lesser General Public License as published
+ *      by the Free Software Foundation, either version 3 of the License, or
+ *      (at your option) any later version.
+ *
+ *      This program is distributed in the hope that it will be useful,
+ *      but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *      MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *      GNU Lesser General Public License for more details.
+ *
+ *      You should have received a copy of the GNU Lesser General Public License
+ *      along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package cc.hyperium.gui;
+
+import cc.hyperium.utils.HyperiumFontRenderer;
+import com.google.common.base.Predicate;
+import com.google.common.base.Predicates;
+import net.minecraft.client.gui.Gui;
+import net.minecraft.client.gui.GuiPageButtonList;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.WorldRenderer;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.util.ChatAllowedCharacters;
+import net.minecraft.util.MathHelper;
+
+public class HyperiumTextField{
```

Comment:
This should probably just extend the normal mc font renderer, amd change requires methods instead of a complete copy.

Inconsistent formatting is another issue

Reply:
I assume you mean the normal GuiTextField, in which case I'll look into that.